instrumentation/aws-sdk/src/main/java/software/amazon/opentelemetry/javaagent/instrumentation/awssdk_v1_11/RequestAccess.java

```diff
@@ -0,0 +1,469 @@
+    if (request == null) {
+      return null;
+    }
+    RequestAccess access = REQUEST_ACCESSORS.get(request.getClass());
+    return invokeOrNull(access.getModelId, request);
+  }
+
+  @Nullable
+  private static String invokeOrNull(@Nullable MethodHandle method, Object obj) {
+    if (method == null) {
+      return null;
+    }
+    try {
+      return (String) method.invoke(obj);
+    } catch (Throwable t) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static <T> T invokeOrNullGeneric(
+      @Nullable MethodHandle method, Object obj, Class<T> returnType) {
+    if (method == null) {
+      return null;
+    }
+    try {
+      return returnType.cast(method.invoke(obj));
+    } catch (Throwable e) {
+      return null;
+    }
+  }
+
+  @Nullable private final MethodHandle getStreamArn;
+  @Nullable private final MethodHandle getAgentId;
+  @Nullable private final MethodHandle getKnowledgeBaseId;
+  @Nullable private final MethodHandle getDataSourceId;
+  @Nullable private final MethodHandle getGuardrailId;
+  @Nullable private final MethodHandle getModelId;
+  @Nullable private final MethodHandle getBody;
+  @Nullable private final MethodHandle getStateMachineArn;
+  @Nullable private final MethodHandle getStepFunctionsActivityArn;
+  @Nullable private final MethodHandle getSnsTopicArn;
+  @Nullable private final MethodHandle getSecretArn;
+  @Nullable private final MethodHandle getLambdaName;
+  @Nullable private final MethodHandle getLambdaResourceId;
+
+  private RequestAccess(Class<?> clz) {
+    getStreamArn = findAccessorOrNull(clz, "getStreamARN", String.class);
+    getAgentId = findAccessorOrNull(clz, "getAgentId", String.class);
+    getKnowledgeBaseId = findAccessorOrNull(clz, "getKnowledgeBaseId", String.class);
+    getDataSourceId = findAccessorOrNull(clz, "getDataSourceId", String.class);
+    getGuardrailId = findAccessorOrNull(clz, "getGuardrailId", String.class);
+    getModelId = findAccessorOrNull(clz, "getModelId", String.class);
+    getBody = findAccessorOrNull(clz, "getBody", ByteBuffer.class);
+    getStateMachineArn = findAccessorOrNull(clz, "getStateMachineArn", String.class);
+    getStepFunctionsActivityArn = findAccessorOrNull(clz, "getActivityArn", String.class);
+    getSnsTopicArn = findAccessorOrNull(clz, "getTopicArn", String.class);
+    getSecretArn = findAccessorOrNull(clz, "getARN", String.class);
+    getLambdaName = findAccessorOrNull(clz, "getFunctionName", String.class);
+    getLambdaResourceId = findAccessorOrNull(clz, "getUUID", String.class);
+  }
+
+  @Nullable
+  private static MethodHandle findAccessorOrNull(
+      Class<?> clz, String methodName, Class<?> returnType) {
+    try {
+      return MethodHandles.publicLookup()
+          .findVirtual(clz, methodName, MethodType.methodType(returnType));
+    } catch (Throwable t) {
+      return null;
+    }
+  }
+
+  @Nullable
+  private static String findNestedAccessorOrNull(Object obj, String... methodNames) {
```

Comment:
I feel like it's better to add some explanation about how java reflection is used here to retrieve the relevant getter method. Or add some comments for `findAccessorOrNull` as well.